src/swarmClient/03_crypto_verify_layer.js

```diff
@@ -111,33 +81,39 @@ module.exports = class Crypto {
             }
         }   
 
-        if(!verify(Buffer.from(signed_bin), Buffer.from(bzn_envelope.getSignature()), bzn_envelope.getSender())) {            
-            this.log && this.log('Bluzelle: signature failed to verify: ' + Buffer.from(bin).toString('hex'));
-        }
 
-        this.onIncomingMsg(bzn_envelope);
+        const sender = format_public(bzn_envelope.getSender());
```

Comment:
This does the format on every message if I understand correctly, which is not ideal in terms of performance.

Reply:
The format_public function is simple string manipulation and does not contain anything cryptographic.

Reply:
True, it's not a huge overhead in comparison, and it's only when crypto is being done which should mostly overshadow it. Microbenchmarking locally says the format operation is ~25% of the time crypto takes (for small messages), which is actually fairly significant but also a deeply flawed testing methodology.

Since this client is designed to operate in a serverless environment where time is money, and usually CPU speed isn't overly quick to begin with, we should avoid adding additional overhead between releases unless it is functionally required (e.g. no crypto -> crypto).

Anyhow, due to the likely low overall impact we can leave this for functional/performance qualification, but a note for future per-message changes.

Reply:
A profiler would be the most appropriate approach to optimization when the time comes.